packages/loader/container-loader/src/container.ts

```diff
@@ -555,6 +556,34 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private get scope() { return this.loader.services.scope; }
     private get codeLoader() { return this.loader.services.codeLoader; }
 
+    /**
+     * {@inheritDoc @fluidframework/container-definitions#IContainer.entryPoint}
+     */
+    public get entryPoint(): Promise<FluidObject | undefined> | undefined {
```

Comment:
I think I've heard it in this team, that getters should not throw. As coded, this one could throw in two different ways: if the container is already closed/disposed (I was surprised to not see that kind of check in other methods here), and if the container emits a `closed` event while the promise below is waiting for `this._context` to stop being undefined.

Would it be better if this was a method? Probably across Container -> Context -> Runtime, if so.

Also curious what others think of the closed/disposed check. A closed container is done for, right? Can't come back? And assuming that's correct, checking `this._lifecycleState` for both options seemed more comprehensive than just checking `this._disposed`.

Reply:
Never mind, made it a method anyway because TS won't let us have an optional accessor, and it started feeling more natural here with these checks and the fact that the return value could change as the container moves through its lifecycle.